client/src/components/deleteModal.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+
+function DeleteModal({id}) {
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  /**
+         * Deletes a record from the database and updates the state of the records.
+         * @param {string} id - The ID of the record to be deleted.
+         * @returns {Promise<void>} A promise that resolves when the record is deleted.
+         */
+  async function deleteRecord(id) {
+    await fetch(`http://localhost:5000/${id}`, {
```

Comment:
It might be okay since the request type is DELETE, but the `await` keyword should be removed since no data is given back when deleting a record.